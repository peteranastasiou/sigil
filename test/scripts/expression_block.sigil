
var b = 1 + { print("A0"); 1 };
print(b);

var arg1 = "hello";
var a = {if arg1 == "hello" { print("A1"); "A2" } else { "A3" }};
print(a);   # should be A2

# Test with locals
var f = fn(){ print({ var b = "b"; var c = "c"; "a"+b+c }); };
f();
