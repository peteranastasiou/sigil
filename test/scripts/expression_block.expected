A0
2
A1
A2
abc